src/Domain/CoordinateMaps/TimeDependent/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@
 # See LICENSE.txt for details.
 
 spectre_target_sources(
-  CoordinateMaps
+  ${LIBRARY}
```

Comment:
I'm a bit undecided if this change is good. Here's an article that I found useful: https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1#avoid-custom-variables-in-the-arguments-of-project-commands. The particular section that the link points to suggests to avoid such temporary variables in CMake to keep things simple. In particular, the variable here is defined in the parent scope. What do you think?